config/config-api/build.gradle

```diff
@@ -17,11 +17,13 @@
 description = 'GoCD Config API'
 dependencies {
   compile project(':util')
-  compile project(':go-plugin-access')
+  compile project(':plugin-metadata-store')
   compile project(':db')
   compile group: 'org.quartz-scheduler', name: 'quartz', version: '1.6.5'
   compile group: 'uk.com.robust-it', name: 'cloning', version: '1.7.3'
   compile group: 'jdom', name: 'jdom', version: '1.0'
+  compile group: 'org.slf4j', name: 'slf4j-api', version: '1.7.21'
+  compile group: 'org.apache.felix', name: 'org.apache.felix.framework', version: '4.2.1'
```

Comment:
So, `config-api` was dependent on `go-plugin-access` which internally depends on `org.apache.felix`. Since I removed dependency on `go-plugin-access`, I had to add an explicit dependency on felix to make compilation work. 
Now, as far as why felix is required by config-api is concerned, then there is `PluginNamespace` class which needs access to `BundleContext`(provided by felix). `PluginNamespace` is used by `ConfigElementImplementationRegistry`. To me it looks like the remnants of the original plugin implementation but I could be wrong. So, one of the tasks for me is to see what is `PluginNamespace` being used for, and if not required then throw it away (different PR).